File: Problems_and_Solutions/day-9/cpp/solution 4.cpp
//Function to push an element into the queue.
void MyQueue:: push(int x)
{
        // Your Code
        QueueNode* cur=new QueueNode(x);
        if(front==NULL)
            front=rear=cur;
        else
        {
            rear->next=cur;
            rear=cur;
        }
}

//Function to pop front element from the queue.
int MyQueue :: pop()
{
        // Your Code 
        if(front==NULL)
            return -1;
        else
        {
            QueueNode* temp=front;
            front=front->next;
            int element=temp->data;
            delete temp;
            if(front==NULL)
                rear=NULL;
            return element;
        }
}

File: Problems_and_Solutions/day-9/Java/Solution4.java
class QueueNode {
    int data;
    QueueNode next;

    QueueNode(int data) {
        this.data = data;
        this.next = null;
    }
}
class Solution4 {
    static QueueNode front, rear;

    public static void push(int a) {
        QueueNode newNode = new QueueNode(a);
        if (front == null) {
            rear = front = newNode;
        } else {
            rear.next = newNode;
            rear = newNode;
        }
    }

    public static int pop() {
        if (front == null) return -1;
        int pop = front.data;
        front = front.next;
        return pop;
    }

    public static void main(String[] args) {
        push(10);
        push(20);
        push(30);
        System.out.println(pop());
        System.out.println(pop());
    }
}

